packages/SwingSet/src/vats/comms/delivery.js

```diff
@@ -236,16 +255,65 @@ export function makeDeliveryKit(state, syscall, transmit, clistKit, stateKit) {
     return { send: target, kernel: true };
   }
 
+  function resolutionCollector() {
+    const resolutions = [];
+    const doneResolutions = new Set();
+
+    function scanSlots(slots) {
+      for (const slot of slots) {
+        const { type } = parseLocalSlot(slot);
+        if (type === 'promise') {
+          const p = state.promiseTable.get(slot);
+          assert(p, X`should have a value for ${slot} but didn't`);
+          if (p.resolved && !doneResolutions.has(slot)) {
+            collect(slot);
+          }
+        }
+      }
+    }
+
+    function collect(lpid) {
+      doneResolutions.add(lpid);
+      const p = state.promiseTable.get(lpid);
+      resolutions.push([lpid, p.rejected, p.data]);
+      scanSlots(p.data.slots);
+    }
+
+    function forPromise(lpid) {
```

Comment:
It's interesting that we don't use the `forPromise` API in this file. Is that because we're assuming that every inbound `dispatch.notify` or comms-side notify will translate N-to-N into an outbound `syscall.resolve` or comms-side notify for each subscriber?

That might be true, I'd have to think about it more.

Reply:
This is here because I did a copy'n'edit from the kernel.  I left `forPromise` in there because I anticipate that we will eventually it, but we just haven't gotten there yet.